exercises/practice/acronym/tests/acronym_test.bal

```diff
@@ -0,0 +1,22 @@
+import ballerina/test;
+
+@test:Config {
+    dataProvider: testCases
+}
+function testFunction(string phrase, string acronym) {
+    test:assertEquals(abbreviate(phrase), acronym);
+}
+
+function testCases() returns map<[string, string]>|error {
+    return {
+        "basic": ["Portable Network Graphics", "PNG"],
+        "lowercase words": ["Ruby on Rails", "ROR"],
+        "punctuation": ["First In, First Out", "FIFO"],
+        "all caps word": ["GNU Image Manipulation Program", "GIMP"],
+        "punctuation without whitespace": ["Complementary metal-oxide semiconductor", "CMOS"],
+        "very long abbreviation": ["Rolling On The Floor Laughing So Hard That My Dogs Came Over And Licked Me", "ROTFLSHTMDCOALM"],
+        "consecutive delimiters": ["Something - I made up from thin air", "SIMUFTA"],
+        "apostrophes": ["Halley's Comet", "HC"],
+        "underscore emphasis": ["The Road _Not_ Taken", "TRNT"]
+    };
+}
```

Comment:
Parameterized tests can be a bit cumbersome:

- Any error usually points to the failing assertion, which makes it slightly harder for the student to spot where the error was
- If we ever upgrade the test runner to version 2, the "test_code" field might just contain the assertion, unless there is a nice way to get the actual value
- Separate test cases are easier to solve in a TDD like way

Reply:
Right, the test runner. I knew there was something I was forgetting.

Ballerina seems to be pretty good at saying which test case failed, even with the dataProvider pattern.

I haven't added "skips" yet, now would be a good time to start.

Reply:
handling skips in the exercise validation script needs to be done first.